### PR TITLE
Add missing `trans` block in usage templates

### PR DESCRIPTION
**Pull request by zerolab**
The usage templates were updated in #9170 / #9279, but the "Field" column header was not translated.

While at it, dropped `valign="top"` from the entity type cell since we don't output anything else in there.

Before | After
-------|---------
before | after

### CODE REVIEW

File: wagtail/documents/templates/wagtaildocs/documents/usage.html [hunk 1]
```diff
@@ -10,7 +10,7 @@
             <thead>
                 <tr>
                     <th class="title">{% trans "Title" %}</th>
-                    <th>Field</th>
+                    <th>{% trans "Field" %}</th>
```
Comment (gasman) [inline wagtail/documents/templates/wagtaildocs/documents/usage.html:13]: Looks like the valign="top" in this file can be removed too.

**Review by gasman — CHANGES_REQUESTED** [review]

File: wagtail/documents/templates/wagtaildocs/documents/usage.html (hunk 1 above)
Comment (zerolab) [reply wagtail/documents/templates/wagtaildocs/documents/usage.html:13]: gah, undid it to make the screenshot and forgot to remove again.
sorted

**Review by gasman — APPROVED** [review]
Thanks, LGTM now!

**Comment by gasman** [comment]
Cherry-picked to 4.1.x in 6ec3ac99c29b4e0ef727831395a43e8ce8eb00e5 / d4f5648cdac5631abe2a2c41ce6f90908be7e6b2.

**Pull request merged**